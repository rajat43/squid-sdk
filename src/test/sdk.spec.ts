import axios from "axios";

import { Squid } from "../index";

jest.mock("axios");

describe("Squid", () => {
  it("should instance Squid as expected", async () => {
    const axiosInstance = (axios.create as jest.Mock).mockReturnValue({
      get: jest.fn()
    });
    const squidSdk = new Squid();

    expect(squidSdk).toMatchSnapshot();
    expect(squidSdk.initialized).toBe(false);
    expect(squidSdk.tokens).toEqual([]);
    expect(squidSdk.chains).toEqual([]);
  });

  describe("setConfig", () => {
    it("should set config as expected", () => {
      const squidSdk = new Squid();

      squidSdk.setConfig({
        executionSettings: {
          infiniteApproval: false
        }
      });

      expect(squidSdk.config).toStrictEqual({
        executionSettings: {
          infiniteApproval: false
        }
      });
    });
  });

  describe("axios module", () => {
    it("get", async () => {
      const axiosInstance = (axios.create as jest.Mock).mockReturnValue({
        get: jest.fn().mockResolvedValue({
          data: { status: true, data: { chains: [] }, tokens: [] }
        })
      });

      const squidSdk = new Squid();

      await squidSdk.init();
      expect(squidSdk.initialized).toEqual(true);
    });
  });
});
