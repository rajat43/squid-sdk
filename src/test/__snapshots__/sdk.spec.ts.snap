// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squid should instance Squid as expected 1`] = `
Squid {
  "axiosInstance": Object {
    "get": [MockFunction],
  },
  "chains": Array [],
  "config": Object {
    "baseUrl": "https://testnet.api.0xsquid.com/",
  },
  "initialized": false,
  "tokens": Array [],
}
`;
